Branch-TestBed/Branch-SDK Debugger Tests/Branch_SDK_Debugger_Tests.m

```diff
@@ -0,0 +1,65 @@
+//
+//  Branch_SDK_Debugger_Tests.m
+//  Branch-TestBed
+//
+//  Created by Qinwei Gong on 2/23/15.
+//  Copyright (c) 2015 Branch Metrics. All rights reserved.
+//
+
+#import <UIKit/UIKit.h>
+#import <XCTest/XCTest.h>
+#import "BNCPreferenceHelper.h"
+#import "BranchServerInterface.h"
+#import "Nocilla.h"
+
+@interface Branch_SDK_Debugger_Tests : XCTestCase {
+    
+@private
+    BNCPreferenceHelper *prefHelper;
+    BranchServerInterface *serverInterface;
+}
+
+@end
+
+@implementation Branch_SDK_Debugger_Tests
+
+- (void)setUp {
+    [super setUp];
+    
+    [[LSNocilla sharedInstance] start];
+    
+    prefHelper = [[BNCPreferenceHelper alloc] init];
+    serverInterface = [[BranchServerInterface alloc] init];
+    serverInterface.delegate = (id<BNCServerInterfaceDelegate>)prefHelper;
+}
+
+- (void)tearDown {
+    [[LSNocilla sharedInstance] clearStubs];
+    [[LSNocilla sharedInstance] stop];
+    
+    [super tearDown];
+}
+
+- (void)testConnectSucceed {
+    stubRequest(@"POST", [BNCPreferenceHelper getAPIURL:@"debug/connect"])
+    .andReturn(200);
+    
+    [serverInterface connectToDebug];
+    
+    XCTAssertTrue([BNCPreferenceHelper getRemoteDebug]);
+}
+
+- (void)testConnectFail {
+    NSDictionary *responseDict = @{@"error": @{@"code": @465, @"message": @"Server not listening"}};
+    NSData *responseData = [BNCServerInterface encodePostParams:responseDict];
+    
+    stubRequest(@"POST", [BNCPreferenceHelper getAPIURL:@"debug/connect"])
+    .andReturn(465)
```

Comment:
It is still important to return a valid status code, as the HTTP handling library that sends these requests that are stubbed still builds up a response as though this was a real request. Say, for example, that you wanted to test a "failure" where Branch return a 299 status (not a valid HTTP code), but say as well that the 299 status of Branch indicated failure. The underlying HTTP lib would say "299 is < 400, so that request must have succeeded" and would fall into your success handler.